pandas/core/dtypes/cast.py

```diff
@@ -375,7 +375,9 @@ def maybe_promote(dtype, fill_value=np.nan):
         if issubclass(dtype.type, np.bool_):
             dtype = np.object_
         elif issubclass(dtype.type, np.integer):
-            dtype = np.float64
+            dtype = np.dtype(np.float64)
+            if not isna(fill_value):
+                fill_value = dtype.type(fill_value)
```

Comment:
Excuse my lack of familiarity but would it be possible for a non-numeric fill_value to hit this condition? If so I think would cause a ValueError

Reply:
No, a few lines up we condition on fill_value being float scalar